### PR TITLE
Make json_report_test test on actual files

### PR DESCRIPTION
**Pull request by nightscape**
We're currently creating a new reporter and along the way noticed that `json_report_test.rb` does not test anything semantically.
This test makes `json_report_test.rb` an integration test that actually runs analyzers on the FakeFS and checks that all sample files appear in the report.

### CODE REVIEW

File: lib/rubycritic/source_control_systems/git.rb [hunk 1]
```diff
@@ -1,32 +1,43 @@
 # frozen_string_literal: true
 
+require 'tty/which'
+
 module RubyCritic
   module SourceControlSystem
     class Git < Base
       register_system
+      GIT_EXECUTABLE = TTY::Which.which('git')
```
Comment (Onumis) [inline lib/rubycritic/source_control_systems/git.rb]: Nice!

File: test/test_helper.rb [hunk 2]
```diff
@@ -18,6 +20,19 @@ def capture_output_streams
...
+    begin
+      config = File.expand_path('..', __dir__)
+      FakeFS::FileSystem.clone(config)
+      Dir.chdir(config)
+      yield
+    ensure
+      FakeFS::FileSystem.clear
+    end
+  end
+end
```
Comment (Onumis) [inline test/test_helper.rb]: :ok_hand:

File: lib/rubycritic/source_locator.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'pathname'
+require 'rubycritic/configuration'
```
Comment (Onumis) [inline lib/rubycritic/source_locator.rb]: Why this require?

**Comment by Onumis** [comment]
Looks great! Just one small question, but overall this is mergable already

File: lib/rubycritic/source_locator.rb (hunk 3 above)
Comment (nightscape) [reply lib/rubycritic/source_locator.rb]: If I remember correctly, the tests failed without this when I ran them outside of `rake` like
```
bundle exec ruby -Ilib:test test/lib/rubycritic/generators/json_report_test.rb 
```
Personally, I also consider it good style to have Ruby files be "self-sufficient", i.e. you can `require` them without having to `require` other files first, but that might be a matter of taste.

**Comment by Onumis** [comment]
Great, thanks!

**Pull request merged**